service/src/main/java/com/commercetools/pspadapter/payone/transaction/IdempotentTransactionExecutor.java

```diff
@@ -8,41 +8,48 @@
 import io.sphere.sdk.payments.TransactionType;
 import io.sphere.sdk.types.CustomFields;
 import io.sphere.sdk.types.Type;
+import org.apache.commons.lang3.StringUtils;
 
+import javax.annotation.Nonnull;
 import java.util.Arrays;
 import java.util.Optional;
-import java.util.function.IntUnaryOperator;
 import java.util.function.Predicate;
 import java.util.stream.IntStream;
 import java.util.stream.Stream;
 
 /**
  * Idempotently executes a Transaction of one Type (e.g. Charge) for a specific PaymentWithCartLike Method.
- *
+ * <p>
  * If no Transaction of that type in State Pending exists, the same PaymentWithCartLike is returned.
  */
 public abstract class IdempotentTransactionExecutor implements TransactionExecutor {
 
     private LoadingCache<String, Type> typeCache;
 
-    public IdempotentTransactionExecutor(final LoadingCache<String, Type> typeCache) {
+    public IdempotentTransactionExecutor(@Nonnull final LoadingCache<String, Type> typeCache) {
         this.typeCache = typeCache;
     }
 
     /**
      * @return The Type that is supported.
      */
+    @Nonnull
     public abstract TransactionType supportedTransactionType();
 
     /**
      * Executes a transaction idempotently.
-     * @param paymentWithCartLike
-     * @param transaction
+     *
+     * @param paymentWithCartLike payment/cart, for which transaction is executed
+     * @param transaction         transaction to execute
      * @return A new version of the PaymentWithCartLike.
      */
     @Override
-    public PaymentWithCartLike executeTransaction(PaymentWithCartLike paymentWithCartLike, Transaction transaction) {
-        if (transaction.getType() != supportedTransactionType()) throw new IllegalArgumentException("Unsupported Transaction Type");
+    @Nonnull
+    public PaymentWithCartLike executeTransaction(@Nonnull PaymentWithCartLike paymentWithCartLike,
+                                                  @Nonnull Transaction transaction) {
+        if (transaction.getType() != supportedTransactionType()) {
```

Comment:
why is the style of of condition different in this `if` condition to the one directly right below?

Reply:
sorry, not clear

Reply:
so the change you did here was add braces to the "then" part of the `if` condition. But you didn't change the below `if` *confused*

Reply:
dbf0c714